### ISSUE TITLE
OIDC : add logout request to IdP

### ISSUE THREAD

**Issue by thopico** [mailcow/mailcow-dockerized#5774]
### Summary

Hi,
I am a OIDC newbie so sorry if it lacks accuracy.
Taking a closer look at [external identity provider feature](https://mailcow.email/posts/2023/mailcow-idp/), I notice there is no logout request sent to the identity provider when logging out from mailcow.
This may be useful to properly and securely end the session.
Thank you for your work !

### Motivation

Security : users may log out from mailcow, but stay logged in the identity provider without notice.
This logout request could be optional.

**Comment by 83E8** [comment]
same Issues with the keycloak authentication source.

**Comment by esackbauer** [comment]
Another request from the forum:
https://community.mailcow.email/d/4945-logout-from-keycloak-doesnt-work/

**Comment by StrategV** [comment]
I agree - we can't logout from keycloak using mailcow inteface now, but this is often necessary!

**Comment by All0ne** [comment]
As long as the fix is in progress/not released, I have a little workaround 🫣 :
by adjusting [data/conf/sogo/custom-sogo.js](https://github.com/mailcow/mailcow-dockerized/blob/master/data/conf/sogo/custom-sogo.js) 

```
// logout function
function mc_logout() {
    fetch("/", {
        method: "POST",
        headers: {
          "Content-Type": "application/x-www-form-urlencoded"
        },
        body: "logout=1"
    }).then(() => window.location.href = 'https://<your-keycloak-domain>/realms/<your-realm>/protocol/openid-connect/logout');
}

```
You can also define a redirect, but you have to allow it in Keycloak ;-) `https://<your-keycloak-domain>/realms/<your-realm>/protocol/openid-connect/logout?redirect_uri=<your-redirect-uri>`

then restarting the containers :) `docker compose restart memcached-mailcow sogo-mailcow`

**Comment by StrategV** [comment]
All0ne, 
Unfortunately:
- redirect_uri has been removed in the keycloak latest versions;
- since there is no token in the link, an extra page with logout confirmation button is appeared;
- logout from SOGo does not affect the Nextcloud in any way (but when you logout from Nextcloud, you also logout from SOGo);
- does not always work (which is strange).
So, the proposed solution only partially solves the problem, but thanks anyway!

**Comment by rikvrede** [comment]
Especially in my org, we have multiple mailboxes that people need to access.
Of course there's workarounds like using private browsing windows, or container tabs in Firefox, but people expect a "log out" button to... you know... log them out. And in our use case they log out so they can log in with a different account.
